### PR TITLE
Create StatefulSet

### PR DESCRIPTION
**Pull request by m-chrzan**
Addresses #4.

### CODE REVIEW

File: operator/cluster-operator.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import (
+	crv1 "github.com/grtl/mysql-operator/pkg/apis/cr/v1"
+	v1beta2 "k8s.io/api/apps/v1beta2"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const MySQLPortNumber = 3306
```
Comment (earlgreyz) [inline operator/cluster-operator.go]: Constant should have a comment or be unexported.

File: operator/cluster-operator.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+func AddCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
```
Comment (earlgreyz) [inline operator/cluster-operator.go]: Exported function should have a documentation comment.

File: operator/cluster-operator.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+func createServiceForCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
+	servicesInterface := kClientset.CoreV1().Services(cluster.ObjectMeta.Namespace)
+
+	newService := serviceForCluster(cluster)
+	_, err := servicesInterface.Create(&newService)
+
+	if err != nil && !apierrors.IsAlreadyExists(err) {
+		panic(err)
```
Comment (earlgreyz) [inline operator/cluster-operator.go]:
```go
logrus.Panic(err)
```
We might also want to debug (or info?) log on an already existing service creation attempt.

File: operator/cluster-operator.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+	createStatefulSetForCluster(cluster, kClientset)
+}
+
+func createServiceForCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
+	servicesInterface := kClientset.CoreV1().Services(cluster.ObjectMeta.Namespace)
+
+	newService := serviceForCluster(cluster)
+	_, err := servicesInterface.Create(&newService)
+
+	if err != nil && !apierrors.IsAlreadyExists(err) {
```
Comment (earlgreyz) [inline operator/cluster-operator.go]: To be consistent let's not put an empty line between function call and the result check.

File: operator/cluster-operator.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+func createStatefulSetForCluster(cluster *crv1.MySQLCluster, kClientset kubernetes.Interface) {
+	statefulSetsInterface := kClientset.AppsV1beta2().StatefulSets(cluster.ObjectMeta.Namespace)
+
+	newStatefulSet := statefulSetForCluster(cluster)
+
+	_, err := statefulSetsInterface.Create(&newStatefulSet)
+
+	if err != nil && !apierrors.IsAlreadyExists(err) {
```
Comment (earlgreyz) [inline operator/cluster-operator.go]: To be consistent let's not put an empty line between function call and the result check.

File: operator/cluster-operator.go [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+				},
+			},
+			Selector: map[string]string{
+				"mysql-cluster": cluster.Spec.Name,
+			},
+		},
+	}
+}
+
+func statefulSetForCluster(cluster *crv1.MySQLCluster) v1beta2.StatefulSet {
```
Comment (earlgreyz) [inline operator/cluster-operator.go]: We might want to simplify this function, for example split the `container` and `pvc` definitions into separate functions

**Comment by m-chrzan** [comment]
This also addresses #8.

**Pull request merged**